src/utils/general.utils.ts

```
@@ -231,8 +231,57 @@ export function formatAmountWithDecimals({ amount, decimals }: { amount: number;
+ * @param amount - number or string to format
+ * @returns formatted string representation of the number
+ */
+export const formatAmount = (amount: string | number): string => {
+    // handle undefined, null, or empty string
+    if (!amount && amount !== 0) return '0'
+
+    // convert amount to number if not already
+    const num = typeof amount === 'string' ? Number(amount) : amount
+
+    // check for NaN after conversion
+    if (isNaN(num)) return '0'
```

Biome (1.9.4)

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)